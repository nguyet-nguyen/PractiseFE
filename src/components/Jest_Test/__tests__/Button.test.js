import React from "react";
import ReactDOM from "react-dom";
import { faker } from "@faker-js/faker";
import render from 'react-test-renderer';
import {act} from 'react-dom/test-utils';
import TestRenderer from 'react-test-renderer';


describe("Unit Test Component render", () => {
    let container;

    beforeEach(() => {
        container = document.createElement('div')
        document.body.appendChild(container)
    })

    afterEach(() => {
        document.body.removeChild(container)
        container = null;
    })

    it("Button component rendering testing",  () => {
        let name = faker.name.firstName() + ' ' + faker.name.lastName()

        act(() => {
            ReactDOM.render(<Button newName={name} />, container)
        })

        const button = container.querySelector('button')
        expect(button.textContent).toBe('Click Me')

        act(() => {
            button.dispatchEvent(new MouseEvent('click', {bubbles: true}));
        })

        expect(button.textContent).toBe('Click ' + name)
    })

 

})