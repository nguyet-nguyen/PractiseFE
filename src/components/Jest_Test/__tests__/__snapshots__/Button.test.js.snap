// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test Component render Button component rendering with SnapShot Part 1 1`] = `
<button>
  Click 
  Me
</button>
`;

exports[`Unit Test Component render Button component rendering with SnapShot Part 1 2`] = `
<button>
  Click 
  Mai Nguyen
</button>
`;
